File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float count_letters(string text);

float count_words(string text);

float count_sentences(string text);

int main(void)
{
    string text = get_string("Insert text here: "); // PROMPT USER FOR TEXT

    float letters = count_letters(text); // COUNT NUMBER OF LETTERS

    float words = count_words(text); // COUNT NUMBER OF WORDS

    float sentences = count_sentences(text); // COUNT NUMBER OF SENTENCES

    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8); // COMPUTE COLEMAN-LIAU INDEX m f f       

    if ((1 <= index) && (index < 16)) // PRINT GRADE LEVEL
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}

float count_letters(string text)
{
    float letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters; // RETURN LETTER COUNT
}

float count_words(string text)
{
    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isblank(text[i]))
        {
            words++;
        }
    }
    return words; // RETURN WORD COUNT
}

float count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences; // RETURN SENTENCE COUNT
}

File: sentimental-readability/readability.py
from cs50 import get_string
import string

text = get_string("Insert text here: ")  # prompt for text input


def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


def count_words(text):
    words = 1
    for i in range(len(text)):
        if text[i].isspace():
            words += 1
    return words


def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1
    return sentences


letters = count_letters(text)  # count letters

words = count_words(text)  # count words

sentences = count_sentences(text)  # count sentences

L = (letters/words)*100
S = (sentences/words)*100
index = round(0.0588 * L - 0.296 * S - 15.8)  # Coleman-Liau index

if 1 <= index and index < 16:
    print("Grade " + str(index))
elif index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
